File: shared/WaiComputerKit/Sources/WaiComputerKit/Audio/RecordingAudioUploadPolicy.swift
import Foundation

public enum RecordingAudioUploadPolicy {
    /// ElevenLabs file STT accepts uploaded files from 100 ms upward.
    /// Keep this as the transport minimum; product-level "too quiet" or
    /// "too short to be useful" decisions should use a separate threshold.
    public static let minimumDurationSeconds: TimeInterval = 0.1

    public static func canUploadFinalizedAudio(
        durationSeconds: TimeInterval?,
        pcmBytesWritten: Int64
    ) -> Bool {
        guard let durationSeconds else { return false }
        return pcmBytesWritten > 0 && durationSeconds >= minimumDurationSeconds
    }
}

File: desktop/WaiComputer.Core/Recording/RecordingAudioUploadPolicy.cs
namespace WaiComputer.Core.Recordings;

/// <summary>
/// Decides whether a finalized recording's audio is eligible for upload, porting
/// the macOS <c>RecordingAudioUploadPolicy</c>. File STT accepts uploaded audio
/// from 100 ms upward; this is the transport floor, not a product-level
/// "too short / too quiet" judgement.
/// </summary>
public static class RecordingAudioUploadPolicy
{
    public const double MinimumDurationSeconds = 0.1;

    public static bool CanUploadFinalizedAudio(double? durationSeconds, long pcmBytesWritten)
        => durationSeconds is { } d && pcmBytesWritten > 0 && d >= MinimumDurationSeconds;
}
